ecosystem/typescript/sdk_v2/src/core/account.ts

```diff
@@ -0,0 +1,168 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import nacl from "tweetnacl";
+import * as bip39 from "@scure/bip39";
+import { AccountAddress } from "./account_address";
+import { Hex } from "./hex";
+import { bytesToHex } from "@noble/hashes/utils";
+import { HexInput } from "../types";
+import { PrivateKey, PublicKey, Signature } from "../crypto/asymmetric_crypto";
+import { derivePath } from "../utils/hd-key";
+import { AuthenticationKey } from "../crypto/authentication_key";
+import { Ed25519PrivateKey, Ed25519PublicKey, Ed25519Signature } from "../crypto/ed25519";
+
+/**
+ * Class for creating and managing account on Aptos network
+ *
+ * Use this class to create accounts, sign transactions, and more.
+ * Note: Creating an account instance does not create the account onchain.
+ */
+export class Account {
+  /**
+   * A private key and public key, associated with the given account
+   */
+  readonly publicKey: PublicKey;
+  readonly privateKey: PrivateKey;
```

Comment:
Note the private key doesn't apply to the MultiEd25519.

This is a little weird where we'd probably want the ability to have an account that is either:

1. A single key account -> which can sign directly there with the private key
2. A non-signable multi-sig account -> which you can verify, but not sign, and won't have a private key.
3. Future will have passkeys or other types of accounts, which also may not have all the private information directly.

Additionally, for generated docs, let's try to put the comments above each item individually.

Let's roll with this for right now, but we need to clean this up probably mid next week to ensure there's a way to handle signatures with non single private key accounts.

Reply:
Yeah, this was the part where I spent the most time thinking on how to optimize the design, in order to be less confusing and complex.

At the end, I feel like if we want to accommodate all the scenarios, this class might start to get a bit complicate very quickly.